tests/Silverback.Integration.Tests/Messaging/Validation/ValidatorConsumerBehaviorTests.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NSubstitute;
+using Silverback.Diagnostics;
+using Silverback.Messaging.Broker;
+using Silverback.Messaging.Broker.Behaviors;
+using Silverback.Messaging.Messages;
+using Silverback.Messaging.Sequences;
+using Silverback.Messaging.Validation;
+using Silverback.Tests.Types;
+using Silverback.Tests.Types.Domain;
+using Xunit;
+
+namespace Silverback.Tests.Integration.Messaging.Validation
+{
+    public class ValidatorConsumerBehaviorTests
+    {
+        [Fact]
+        public async Task HandleAsync_InvalidMessage_ExceptionIsThrown()
+        {
+            var endpoint = TestProducerEndpoint.GetDefault();
+            endpoint.MessageValidationMode = MessageValidationMode.LogWarning;
+            var headers = new[]
+            {
+                new MessageHeader("x-message-type", typeof(TestMessageValidation).AssemblyQualifiedName)
```

Comment:
I don't think you need this, you should provide the message in a deserialized state already, since you aren't going to send it through the whole pipeline and nothing will deserialize it.